server/src/main/java/org/apache/cassandra/sidecar/livemigration/LiveMigrationInstanceMetadataUtil.java

```diff
@@ -0,0 +1,193 @@
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.LOCAL_SYSTEM_DATA_FILE_DIR_PLACEHOLDER;
+import static org.apache.cassandra.sidecar.livemigration.LiveMigrationPlaceholderUtil.SAVED_CACHES_DIR_PLACEHOLDER;
+
+/**
+ * Utility class for having all {@link InstanceMetadata} related helper functions related to
+ * Live Migration in one place.
+ */
+@SuppressWarnings("ConstantValue")
+public class LiveMigrationInstanceMetadataUtil
+{
+
+    /**
+     * List of directories Live Migration considers to transfer between source and destination.
+     */
+    public static List<String> dirsToCopy(InstanceMetadata instanceMetadata)
+    {
+        List<String> dirsToCopy = new ArrayList<>();
+        dirsToCopy.add(instanceMetadata.hintsDir());
+        dirsToCopy.add(instanceMetadata.commitlogDir());
+        if (instanceMetadata.savedCachesDir() != null)
+        {
+            dirsToCopy.add(instanceMetadata.savedCachesDir());
+        }
+        if (instanceMetadata.cdcDir() != null)
+        {
+            dirsToCopy.add(instanceMetadata.cdcDir());
+        }
+        if (instanceMetadata.localSystemDataFileDir() != null)
+        {
+            dirsToCopy.add(instanceMetadata.localSystemDataFileDir());
+        }
+        dirsToCopy.addAll(instanceMetadata.dataDirs());
+        return dirsToCopy;
+    }
+
+    /**
+     * Returns map of directory that can be copied and the index to use while constructing
+     * url for file transfer.
+     */
+    public static Map<String, String> dirPathPrefixMap(InstanceMetadata instanceMetadata)
+    {
+        Map<String, String> dirIndexMap = new Hashtable<>();
+        dirIndexMap.put(instanceMetadata.hintsDir(), LIVE_MIGRATION_HINTS_DIR_PATH + "/0");
+        dirIndexMap.put(instanceMetadata.commitlogDir(), LIVE_MIGRATION_COMMITLOG_DIR_PATH + "/0");
+        if (instanceMetadata.savedCachesDir() != null)
+        {
+            dirIndexMap.put(instanceMetadata.savedCachesDir(), LIVE_MIGRATION_SAVED_CACHES_DIR_PATH + "/0");
+        }
+        if (instanceMetadata.cdcDir() != null)
+        {
+            dirIndexMap.put(instanceMetadata.cdcDir(), LIVE_MIGRATION_CDC_RAW_DIR_PATH + "/0");
+        }
+        if (instanceMetadata.localSystemDataFileDir() != null)
+        {
+            dirIndexMap.put(instanceMetadata.localSystemDataFileDir(), LIVE_MIGRATION_LOCAL_SYSTEM_DATA_FILE_DIR_PATH + "/0");
+        }
+        for (int i = 0; i < instanceMetadata.dataDirs().size(); i++)
+        {
+            dirIndexMap.put(instanceMetadata.dataDirs().get(i), LIVE_MIGRATION_DATA_FILE_DIR_PATH + "/" + i);
+        }
+        return dirIndexMap;
+    }
+
+    /**
+     * Returns map of directory and set of placeholders that represent the directory.
+     */
+    public static Map<String, Set<String>> dirPlaceHoldersMap(InstanceMetadata instanceMetadata)
+    {
+        Map<String, Set<String>> placeholderMap = new Hashtable<>();
```

Comment:
Can we use a hashmap here?
```suggestion
        Map<String, Set<String>> placeholderMap = new HashMap<>();
```

Reply:
HashMap allows null key and values. placeholderMap should not have either null key or null value.

Reply:
same comment as above for the usage of Hashtable

Reply:
Updated it.